momento-redis/sorted_set_commands.go

```diff
@@ -250,3 +288,161 @@ func (m *MomentoRedisClient) ZDiffStore(ctx context.Context, destination string,
+	case *responses.SortedSetFetchMiss:
+		// empty array is default
+	}
+
+	return resp
+}
+
+type MomentoZRangeBy struct {
+	MinScore *float64
+	MaxScore *float64
+	Offset   *uint32
+	Count    *uint32
+}
+
+func validateFetchByScoreOptions(opt *ZRangeBy) MomentoZRangeBy {
+	mZRange := &MomentoZRangeBy{}
+	if opt.Count < 0 {
+		panic(UnsupportedOperationError("Negative count is not supported by Momento"))
+	}
+	if opt.Count > math.MaxUint32 {
+		panic(UnsupportedOperationError("Count exceeds max supported integer value by Momento"))
+	}
+	if opt.Offset < 0 {
+		panic(UnsupportedOperationError("Negative offset is not supported by Momento"))
+	}
+	if opt.Offset > math.MaxUint32 {
+		panic(UnsupportedOperationError("Offset exceeds max supported integer value by Momento"))
+	}
+
+	// this is yet another special case. Go-Redis doesn't specify Count as a pointer, so we cannot reference
+	// it for a null check. Go-lang by default treats integers as 0. So a Redis customer can not provide the
+	// count and still have all the elements in the sorted-set even if Go-Lang treats it as 0. This is because
+	// Redis simply streams the entire command as a string for the params provided by the client and excludes
+	// the default 0 count. In case of Momento, we don't want to send this default value as it will always
+	// result in returning an empty set of values.
+	if opt.Count != 0 {
+		count := uint32(opt.Count)
+		mZRange.Count = &count
+	}
+
+	offset := uint32(opt.Offset)
+	mZRange.Offset = &offset
+
+	// if client provided +inf, we don't pass anything to Momento as that's the default UnboundedMax
+	// the length check is present because an empty string is treated as 0 and will be assigned as such later
+	if opt.Min != "-inf" && len(opt.Min) > 0 {
+		if strings.HasPrefix(opt.Min, "(") {
+			panic(UnsupportedOperationError("Momento currently does not support exclusion of scores. Please omit ( from your" +
+				" request and retry. Reach out to us at Discord https://discord.com/invite/3HkAKjUZGq) or e-mail us at " +
+				"support@momentohq.com if you want such capabilities!"))
+		}
+		val, err := strconv.ParseFloat(opt.Min, 64)
+		if err != nil {
+			panic("Cannot parse string MinScore into float, MinScore: " + opt.Min)
+		}
+		mZRange.MinScore = &val
+	}
+
+	// if client provided +inf, we don't pass anything to Momento as that's the default UnboundedMax
+	// the length check is present because an empty string is treated as 0 and will be assigned as such later
+	if opt.Max != "+inf" && len(opt.Max) > 0 {
+		if strings.HasPrefix(opt.Max, "(") {
+			panic(UnsupportedOperationError("Momento currently does not support exclusion of scores. Please omit ( from your" +
+				" request and retry. Reach out to us at Discord https://discord.com/invite/3HkAKjUZGq) or e-mail us at " +
+				"support@momentohq.com if you want such capabilities!"))
+		}
+		val, err := strconv.ParseFloat(opt.Max, 64)
+		if err != nil {
+			panic("Cannot parse string MaxScore into float")
```

Comment:
Can you go ahead and output the value in the message here like you did with `MinScore` above?